go/vt/vtgate/planbuilder/operators/phases.go

```diff
@@ -101,13 +101,18 @@ type phaser struct {
 }
 
 func (p *phaser) next(ctx *plancontext.PlanningContext) Phase {
-	for phas := p.current; phas < DONE; phas++ {
-		if phas.shouldRun(ctx.SemTable.QuerySignature) {
-			p.current = p.current + 1
-			return phas
+	for {
```

Comment:
the old code for this was wrong, and we were hitting some phases twice. didn't really make a noticeable difference in plan outputs, but it took longer than needed to run.

Reply:
good catch